### PR TITLE
feat(ci): run e2e tests against postgres/mysql backends on demand

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

When there's changes under `src/meta`, a new label `ci/run-e2e-test-other-backends` will be attached to the PR. Then, end-to-end tests against postgres/mysql meta backends will be triggered.

Developers can also manually attach the label to trigger the tests.

**NOTE**: This PR only enables Postgres backend, as I found several bugs blocking the tests under the MySQL backend. Will address them and enable it in next PRs.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: ci/workflows/pull-request.yml [hunk 1]
```diff
@@ -790,6 +790,38 @@ steps:
     timeout_in_minutes: 30
     retry: *auto-retry
 
+  - label: "end-to-end test (parallel, postgres backend)"
```
Comment (kwannoel) [inline ci/workflows/pull-request.yml:793]: Shall we just run e2e-test against postgres as the default option?
For the others we can run in `main-cron`

Comment (kwannoel) [reply ci/workflows/pull-request.yml:793]: Otherwise it may be expensive to run all combination, across all our e2e test types.

Comment (kwannoel) [reply ci/workflows/pull-request.yml:793]: Nvm just saw the labeller workflow.

**Comment by BugenZhao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19152?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#19152** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19152?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#19155** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19155?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BugenZhao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/meta/model/migration/src/m20241025_062548_singleton_vnode_count.rs [hunk 2]
```diff
@@ -6,22 +6,21 @@ pub struct Migration;
...
+
         // Fill vnode count with 1 for singleton tables.
         manager
             .exec_stmt(
                 UpdateStatement::new()
                     .table(Table::Table)
                     .values([(Table::VnodeCount, Expr::value(1))])
-                    .and_where(
-                        Expr::col(Table::DistributionKey)
-                            .cast_as(Alias::new("varchar"))
```
Comment (BugenZhao) [inline src/meta/model/migration/src/m20241025_062548_singleton_vnode_count.rs:17]: MySQL does not support casting to `VARCHAR`. 🤡🤡🤡🤡🤡🤡🤡🤡🤡🤡

**Review by lmatz — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]

File: ci/workflows/pull-request.yml [hunk 3]
```diff
@@ -790,6 +790,40 @@ steps:
...
+          run: pg-mysql-backend-test-env
+          config: ci/docker-compose.yml
+          mount-buildkite-agent: true
+      - ./ci/plugins/upload-failure-logs
+    timeout_in_minutes: 30
+    retry: *auto-retry
+
+  - label: "end-to-end test (mysql backend)"
+    command: "ci/scripts/e2e-test.sh -p ci-dev -m ci-3streaming-2serving-3fe-mysql-backend"
+    if: false
```
Comment (BugenZhao) [inline ci/workflows/pull-request.yml:811]: Why it still run even specifying `if: false`? 😄

Comment (BugenZhao) [reply ci/workflows/pull-request.yml]: if: "false" works

**Pull request merged**